### PR TITLE
 feat: New secp system cells (with dep group cell)

### PR DESCRIPTION
**Pull request by TheWaWaR**
- [x] Fixed bench secp cell deps
- [x] Add integration test for secp dep group

### CODE REVIEW

File: util/types/src/extension/capacity.rs [hunk 1]
```diff
@@ -37,7 +37,10 @@ impl packed::CellOutput {
 }
 
 impl packed::CellOutputBuilder {
-    pub fn reset_capacity(self, data_capacity: Capacity) -> CapacityResult<Self> {
+    pub fn build_exact_capacity(
```
Comment (yangby-cryptape) [inline util/types/src/extension/capacity.rs]: :+1:

File: Makefile [hunk 2]
```diff
@@ -89,7 +89,7 @@ security-audit: ## Use cargo-audit to audit Cargo.lock for crates with security
 
 bench-test:
 	echo "Fix the secp256k1 bench test and reenable it"
```
Comment (u2) [inline Makefile]: remove the echo message.

Comment (TheWaWaR) [reply Makefile]: Oops

File: docs/hashes.toml [hunk 3]
```diff
@@ -1,42 +1,72 @@
...
 
 [ckb_staging]
-genesis = "0xebdca80c83260cee3013a210372c4d934631f948bbd044eafdb448d6564cc487"
-cellbase = "0x3c1efe9a3c9ca1a008a5a52be9084a38faa88e8b8fedbc247284c847e519d66d"
+genesis = "0x6853d356bec5eede3c7e6dccba18c1008570930e7ecca8282d6074e856777d3e"
+cellbase = "0x3ff78c6d260ca3a900d6bb9da81784cfe9cd9e3d480a135483b64067fb4a4616"
 
 [[ckb_staging.system_cells]]
-path = "Bundled(specs/cells/secp256k1_blake160_sighash_all)"
+path = "Bundled(specs/cells/secp256k1_data)"
```
Comment (doitian) [inline docs/hashes.toml]: Please do not change the position of the existing cells.

1. secp256k1_blake160_sighash_all
2. dao
3. secp256k1_data
4. secp256k1_ripemd160_sha256_sighash_all

Comment (doitian) [reply docs/hashes.toml]: TODO: we should include the second transaction hash in hashes.toml

File: spec/src/lib.rs [hunk 4]
```diff
@@ -229,6 +257,109 @@ impl Genesis {
             .build();
         Ok(tx)
     }
+
+    fn build_dep_group_transaction(
```
Comment (doitian) [inline spec/src/lib.rs]: TODO: make it configurable via chain spec.

File: docs/hashes.toml (hunk 3 above)
Comment (TheWaWaR) [reply docs/hashes.toml]: ok

File: spec/src/lib.rs [hunk 5]
```diff
@@ -30,6 +36,11 @@ use std::sync::Arc;
 
 pub mod consensus;
 
+// Just a random secp256k1 secret key for dep group input cell's lock
+const SPECIAL_CELL_PRIVKEY: H256 =
+    h256!("0xd0c5c1e2d5af8b6ced3c0800937f996c1fa38c29186cade0cd8b5a73c97aaca3");
+const SPECIAL_CELL_CAPACITY: Capacity = capacity_bytes!(500);
```
Comment (doitian) [inline spec/src/lib.rs]: The capacity can be calculated if we make the dep group creation configuration. See #1435

**Pull request merged**